sdk/include/opentelemetry/sdk/logs/logger_provider_factory.h

```diff
@@ -0,0 +1,66 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0/
+
+#pragma once
+
+#ifdef ENABLE_LOGS_PREVIEW
+
+#  include <memory>
+#  include <vector>
+
+#  include "opentelemetry/logs/logger_provider.h"
+#  include "opentelemetry/nostd/shared_ptr.h"
+#  include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#  include "opentelemetry/sdk/logs/logger.h"
+#  include "opentelemetry/sdk/logs/logger_context.h"
+#  include "opentelemetry/sdk/logs/processor.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+
+/**
+ * Factory class for LoggerProvider.
+ */
+class LoggerProviderFactory
+{
+public:
+  /**
+   * Create a LoggerProvider.
+   */
+  static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create(
+      std::unique_ptr<LogProcessor> &&processor);
+
+  /**
+   * Create a LoggerProvider.
+   */
+  static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create(
+      std::unique_ptr<LogProcessor> &&processor,
+      const opentelemetry::sdk::resource::Resource &resource);
+
+  /**
+   * Create a LoggerProvider.
+   */
+  static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create(
+      std::vector<std::unique_ptr<LogProcessor>> &&processors);
+
+  /**
+   * Create a LoggerProvider.
+   */
+  static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create(
+      std::vector<std::unique_ptr<LogProcessor>> &&processors,
+      const opentelemetry::sdk::resource::Resource &resource);
+
+  /**
+   * Create a LoggerProvider.
+   */
+  static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create(
+      std::shared_ptr<sdk::logs::LoggerContext> context);
+};
+
```

Comment:
Should we also have
```cpp
static std::unique_ptr<opentelemetry::logs::LoggerProvider> Create()
```
As it is possible to add `LogProcessor` instance after creating LoggerProvider?

Reply:
Create() returns the API abstraction, here` opentelemetry::logs::LoggerProvider`.

This class in the API does not have a way to add log processors,
only the SDK class `opentelemetry::sdk::logs::LoggerProvider` can.

The whole point it to hide the SDK class implementation,
which as a side effect also hides the extra methods the SDK can provide.

By using SDK classes directly, the application owner can:
- put all the log processors in a vector, and create a logger provider
- create a logger provider and add log processors one by one, calling `AddProcessor()`

By using the `Factory::Create()` builder pattern, only the first case is supported.